docker-compose.yml

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+services:
+  # TODO: Add the Dockerfile for the FAISS Index service
+
+  api:
+    build:
+      context: ./backend
+      dockerfile: Dockerfile.api
+    ports:
+      - "8000:8000"
+    volumes:
+      - .:/backend/app
+
+  ui:
+    build:
+      context: ./frontend
+      dockerfile: Dockerfile.ui
+    ports:
+      - "5173:5173"
+    volumes:
+      - .:/frontend/app
```

Comment:
Here you're mounting the source code volume, but then also copying it in the dockerfile.ui:22. 

I think it should still work, but just double check it's doing what you expect. Do any changes in the code locally trigger a UI rebuild in the docker container? 

Same question for the API